### PR TITLE
M/geo shapes docs

### PR DESCRIPTION
**Pull request by mfelsche**

### CODE REVIEW

File: docs/sql/data_types.txt [hunk 1]
```diff
@@ -170,6 +170,74 @@ Create table example::
...
+
+            Supported units are ``inch`` (``in``), ``yard`` (``yd``), ``miles`` (``mi``),
+            ``kilometers`` (``km``), ``meters``(``m``), ``centimeters`` (``cm``), ``millimeters`` (``mm``).
+
+:tree: (Default: ``geohash``) Internal data structure used for indexing.
+       Available trees are ``geohash`` and ``quadtree``.
+       Both are prefix trees representing the earth as grid layers with increasing precision.
+       ``quadtree`` uses a `Quadtree`_ to represent grids and ``geohash`` uses `Geohash`_.
+       The main difference is that the ``quadtree`` supports higher precision
+       than the ``geohash`` tree.
```
Comment (bogensberger) [inline docs/sql/data_types.txt]: i think the tree should be explained more detailed. How does it affect index size? Advantages and disadvantages of quadtree / geohash

File: docs/sql/data_types.txt [hunk 2]
```diff
@@ -170,6 +170,74 @@ Create table example::
...
+----------
+
+A geo_shape Column is defined using the following Syntax::
+
+    <columnName> GEO_SHAPE [ (<parameter>=<value> [, ...]) ]
+
+Where the following parameters are supported:
+
+
+:precision: (Default: ``50m``) Define the precision of the indexed shape used for querying.
```
Comment (dobe) [inline docs/sql/data_types.txt]: so default is 50

File: docs/sql/scalar.txt [hunk 3]
```diff
@@ -464,7 +462,42 @@ casting from strings to geo point and geo shapes::
...
+    +----------------------+
+    | intersect | disjoint |
+    +----------------------+
+    | TRUE      | FALSE    |
+    +----------------------+
+    SELECT 1 row in set (... sec)
+
+Due to a limitation on the :ref:`geo_shape_data_type` datatype
+this function can only be used within the ``WHERE`` clause
+and in ``SELECT`` expressions.
```
Comment (dobe) [inline docs/sql/scalar.txt]: we need it also as scalar

**Comment by mfussenegger** [comment]
ack

**Pull request merged**